instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py

```diff
@@ -56,7 +56,7 @@
 
 # A key to a context variable to avoid creating duplicate spans when instrumenting
 # both, Session.request and Session.send, since Session.request calls into Session.send
-_SUPPRESS_REQUESTS_INSTRUMENTATION_KEY = "suppress_requests_instrumentation"
+_SUPPRESS_HTTP_INSTRUMENTATION_KEY = "suppress_http_instrumentation"
```

Comment:
Curious as to why we are merging these into one context variable now?

Reply:
`requests` uses `urllib3` internally. If the instrumentations for both libraries are installed two spans equal spans would be created. So the common suppression key is to avoid this double tracing.

Reply:
Applies to `urllib` as well?

Reply:
`urllib` is actually separate but all three libraries internally use the `http.client` package. So in case there is a `http.client` instrumentation in the future i thought it would make sense to have a common suppression key for that.

Reply:
Does that mean if today, someone does not want to track `requests` but wants to track `urllib`, they wouldn't be able to?

Reply:
I think that the supression variables were only meant to avoid double tracing and not for users to supress spans for one specific library. Like for the `requests` library where a call to `Session.request` would create another span because it calls into `Session.send` which is also instrumented.

To generally avoid tracing for example the `requests` library one would just not install the instrumentation or simply call `RequestsInsrumentor().uninstrument()`. I doubt anyone would use the `Context` for that since the supression variable is then required to be in every context instance to work reliably.

To supress spans for a single call/request using the `Context` probably makes sense but for the `requests`, `urllib3` example (not using `urllib` here since it does not share a code path with `requests`) it wouldn't make much of a difference since in the end both libraries are producing the same kind of span (with even the same attributes).

If we really want to support this i guess we would need 3 supression variables: one to just supress the library (e.g. requests), one to supress the traced aspect (e.g. http) and one to supress tracing alltogether. One could then just set the supression variables for the libraries they do not want to track.

Reply:
I'm fine with the current changes since the supression variables were meant to only be used internally.